File: _2020/python/day1.py
import itertools

from _2020.python.utils import load_puzzle_input, profile

puzzle_input = list(map(int, load_puzzle_input("day1").split("\n")))


@profile
def part1():
    for a, b in itertools.combinations(puzzle_input, 2):
        if a + b == 2020:
            return a * b


@profile
def part2():
    for a, b, c in itertools.combinations(puzzle_input, 3):
        if a + b + c == 2020:
            return a * b * c


if __name__ == "__main__":
    part1()
    part2()

File: _2020/go/day1.go
package main

import (
	"fmt"
	"strconv"
	"strings"
	"utils"
)

func parseInput() []int {
	data := utils.LoadPuzzleInput("day1")
	input_strings := strings.Fields(data)
	integers := make([]int, len(input_strings))
	for ind, str := range input_strings {
		i, _ := strconv.Atoi(str)
		integers[ind] = i
	}
	return integers
}

func day1part1() string {
	integers := parseInput()
	value := 0
	for _, a := range integers {
		for _, b := range integers {
			if a+b == 2020 {
				value = a * b
				break
			}
		}
	}
	return fmt.Sprintf("%v", value)
}

func day1part2() string {
	integers := parseInput()
	value := 0
	for _, a := range integers {
		for _, b := range integers {
			for _, c := range integers {
				if a+b+c == 2020 {
					value = a * b * c
					break
				}
			}
		}
	}
	return fmt.Sprintf("%v", value)
}

func day1main() {
	utils.Profile(day1part1)
	utils.Profile(day1part2)
}
